distribution/tools/plugin-cli/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -324,9 +325,16 @@ boolean urlExists(Terminal terminal, String urlString) throws IOException {
     /** Downloads a zip from the url, into a temp file under the given temp dir. */
     // pkg private for tests
     @SuppressForbidden(reason = "We use getInputStream to download plugins")
-    Path downloadZip(Terminal terminal, String urlString, Path tmpDir) throws IOException {
+    Path downloadZip(Terminal terminal, String urlString, Path tmpDir, Path pluginsDir) throws IOException {
         terminal.println(VERBOSE, "Retrieving zip from " + urlString);
         URL url = new URL(urlString);
+        if (url.getProtocol().equals("file")) {
+            Path pluginsFile = Paths.get(url.getFile());
+            if (pluginsFile.startsWith(pluginsDir)) {
+                throw new IllegalStateException("Installation failed! " +
+                    "Make sure `plugins` directory doesn't contain the plugin distribution! " );
```

Comment:
I think we can have a more actionable error message here like: `"the plugins directory [" + pluginsDir + "] can not contain the plugin distribution [" + pluginsFile + "]; move the distribution to an alternate location"`